### ISSUE TITLE
Inconsistence in /wallet-api/auth/create

### ISSUE THREAD

**Issue by alegomes** [walt-id/waltid-identity#71]
Version tested: v1.0.2401181053-SNAPSHOT

I think `/wallet-api/auth/create` endpoint could be more specific and consistent in the failed response messages. 

Take a look at these four scenarios. They all demonstrate the API call with some missing field. However, the answers vary from one call to another.

# Scenario 1. `type` field omitted.
Request
```
curl -X 'POST' \
  'http://localhost:4545/wallet-api/auth/create' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "Max Mustermann",
  "email": "user@email.com",
  "password": "password"}'
```
Response
```
{"exception":"true","status":"Bad Request","code":"400","message":"Polymorphic serializer was not found for missing class discriminator ('null')\nJSON input: {\"name\":\"Max Mustermann\",\"email\":\"user@email.com\",\"password\":\"password\"}"}
```

# Scenario 2. `password` field omitted
Request
```
curl -X 'POST' \
  'http://localhost:4545/wallet-api/auth/create' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "Max Mustermann",
  "email": "user@email.com",
  "_password": "password",
  "type": "email"
}'
```
Response
```
{"exception":"true","status":"Bad Request","code":"400","message":"Field 'password' is required for type with serial name 'email', but it was missing at path: $"}
```
# Scenario 3. `email` field omitted
Request
```
curl -X 'POST' \
  'http://localhost:4545/wallet-api/auth/create' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "Max Mustermann",
  "_email": "user@email.com",
  "password": "password",
  "type": "email"
}'
```
Response
```
{"exception":"true","status":"Bad Request","code":"400","message":"Field 'email' is required for type with serial name 'email', but it was missing at path: $"}

```

# Scenario 4. `name` field omitted
Request
```
curl -X 'POST' \
  'http://localhost:4545/wallet-api/auth/create' \
  -H 'accept: */*' \
  -H 'Content-Type: application/json' \
  -d '{
  "_name": "Max Mustermann",
  "email": "user@email.com",
  "password": "password",
  "type": "email"
}'
```
Response
```
{"exception":"true","status":"Internal Server Error","code":"500","message":"Could not register user"}
```

**Comment by alegomes** [comment]
Not stale, as explained in #72

**Comment by SuperBatata** [comment]
should be fixed with this pr : #597
